SonarQube.Common/ProcessRunner.cs

```diff
@@ -64,6 +87,16 @@ public bool Execute(string exeName, string args, string workingDirectory, int ti
                 WorkingDirectory = workingDirectory
             };
 
+            if (envVariables != null)
+            {
+                foreach (KeyValuePair<string, string> envVariable in envVariables)
+                {
+                    if (!String.IsNullOrEmpty(envVariable.Value))
+                    {
+                        psi.EnvironmentVariables.Add(envVariable.Key, envVariable.Value);
```

Comment:
I thought about adding unit tests for the ProcessRunner but realized I'd have to create a new executable on the fly (or check one in) which can be done but is rather involved as the logic in not so complex.